src/couch_quickjs/src/couch_quickjs_scanner_plugin.erl

```diff
@@ -182,11 +183,12 @@ process_ddoc(#st{} = St, DbName, #doc{} = DDoc0) ->
             catch
                 throw:{validate, Error} ->
                     Meta = #{sid => SId, db => DbName, ddoc => DDocId},
-                    ?WARN("ddoc validation failed ~p", [Error], Meta),
-                    St1;
+                    validation_warning("ddoc validation failed ~p", Error, Meta),
+                    start_or_reset_procs(St1);
```

Comment:
Why do we call `start_or_reset_procs(St1)` again?
`start_or_reset_sm_proc/1` and `start_or_reset_qjs_proc/1` are recursive functions.
They should return the state with some defined process.

Reply:
We're guarding for the case when the the js processes might have died or crashed, so we want to restart them before returning, if the processes are alive, it's a no-op. So it doesn't hurt to check again at the end of the block.

Reply:
But you're right we can probably skip resetting them again as every other block that wants to use them will end up resetting them first.

Reply:
Updated and removed the extra start_or_reset_procs(St1) in the catch clauses